samples/bluetooth/direction_finding_connectionless_rx/src/main.c

```diff
@@ -0,0 +1,373 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stddef.h>
+#include <errno.h>
+#include <zephyr.h>
+
+#include <sys/printk.h>
+#include <sys/byteorder.h>
+#include <sys/util.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/direction.h>
+
+#define DEVICE_NAME CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LEN (sizeof(DEVICE_NAME) - 1)
+#define NAME_LEN 30
+#define TIMEOUT_SYNC_CREATE K_SECONDS(10)
+
+static struct bt_le_per_adv_sync_param sync_create_param;
+static struct bt_le_per_adv_sync *sync;
+static bt_addr_le_t per_addr;
+static bool per_adv_found;
```

Comment:
Added volatile to `per_adv_found`. Access to this variable may be optimized out in case of 03 optimizations are enabled.
The variable is cleared in main thread and read/updated in BT receive thread. When BT receive thread runs the main thread should be waiting for `sem_per_adv` to be signaled.

`scan_enabled` is used only in main thread context.